src/Rector/Class_/PreferPHPUnitSelfCallRector.php

```diff
@@ -95,8 +98,17 @@ public function refactor(Node $node): ?Node
                 return null;
             }
 
-            $hasChanged = true;
-            return $this->nodeFactory->createStaticCall('self', $methodName, $node->getArgs());
+            $classReflection = $this->reflectionResolver->resolveClassReflectionSourceObject($node);
```

Comment:
```suggestion
            $classReflection = $this->reflectionResolver->resolveClassReflection($node);

Reply:
After some thinking, I think the static check only make sense if method exactly from `PHPUnit\Framework\TestCase`, that means `ClassReflection` need to be pulled from ReflectionProvider:

```php
if ($this->reflectionProvider->hasClass('PHPUnit\Framework\TestCase')) {

    $classReflection = $this->reflectionProvider->getClass('PHPUnit\Framework\TestCase');

// not has method, just skip changing...
 if (! $classReclection->hasNativeMethod($methodName)) {
    return null;
}
```

Reply:
But if I implement my own static `assert` methods in subclasses of `PHPUnit\Framework\TestCase` it would be nice if they were treated the same way as PHPUnits `assert` methods.

Reply:
Ok :), I was thinking the original rule feature was to change to static the `assert*` method, but only from `PHPUnit\Framework\TestCase` object, 

but to keep the behavior, I am fine with it :),

Just change `resolveClassReflectionSourceObject` to `resolveClassReflection` as it validated from `this` variable early.

Reply:
Ok, I have changed that now. 👍 

Thanks for guiding me through the Rector internals. 😄